### PR TITLE
fix(@angular-devkit/build-angular): fixes optimizeChunkAssets is deprecated in webpack 5

### PR DESCRIPTION
**Pull request by valorkin**
should fix 83 warnings `[DEP_WEBPACK_COMPILATION_OPTIMIZE_CHUNK_ASSETS] DeprecationWarning: optimizeChunkAssets is deprecated (use Compilation.hook.processAssets instead and use one of Compilation.PROCESS_ASSETS_STAGE_* as stage option)`

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts [hunk 1]
```diff
@@ -8,25 +8,49 @@
...
+
 function hook(
   compiler: Compiler,
-  action: (
-    compilation: compilation.Compilation,
-    chunks: Iterable<compilation.Chunk>,
-  ) => Promise<void>,
+  action: (compilation: compilation.Compilation, assetsURI: string[]) => Promise<void>,
 ) {
   compiler.hooks.compilation.tap('optimize-css-webpack-plugin', (compilation) => {
```
Comment (alan-agius4) [inline packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts:26]:
```suggestion
  compiler.hooks.compilation.tap(PLUGIN_NAME, (compilation) => {
```

File: packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts [hunk 2]
```diff
@@ -8,25 +8,49 @@
...
-    );
+    if (isWebpackFiveOrHigher()) {
+      // webpack 5 migration "guide"
+      // https://github.com/webpack/webpack/blob/07fc554bef5930f8577f91c91a8b81791fc29746/lib/Compilation.js#L527-L532
+      // TODO_WEBPACK_5 const stage = Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE;
+      const stage = 100;
+      // tslint:disable-next-line: no-any
+      (compilation.hooks as any)
+        .processAssets.tapPromise({name: PLUGIN_NAME, stage}, (assets: Record<string, Source>) => {
+          return action(compilation, Object.keys(assets || {}));
```
Comment (alan-agius4) [inline packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts:35]: Based on the type above, assets cannot be null/undefined therefore `|| {}` is redundant.

Comment (valorkin) [reply packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts:35]: I don't trust webpack5 types, but your are right, this one is `{}` at least 
https://github.com/webpack/webpack/blob/07fc554bef5930f8577f91c91a8b81791fc29746/lib/Compilation.js#L743

File: packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts (hunk 1 above)
Comment (valorkin) [reply packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts]: applied

File: packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts (hunk 2 above)
Comment (valorkin) [reply packages/angular_devkit/build_angular/src/webpack/plugins/optimize-css-webpack-plugin.ts]: done

**Review by alan-agius4 — APPROVED** [review]
👍  thanks LGTM

**Pull request merged**